projects/igniteui-angular/migrations/update-15_1_0/index.ts

```diff
@@ -3,15 +3,79 @@ import {
     SchematicContext,
     Tree
 } from '@angular-devkit/schematics';
+import { Element } from '@angular/compiler';
+import { nativeImport } from '../common/import-helper.js';
 import { Options } from '../../schematics/interfaces/options';
 import { BoundPropertyObject, InputPropertyType, UpdateChanges } from '../common/UpdateChanges';
+import { FileChange, findElementNodes, getAttribute, getSourceOffset, parseFile } from '../common/util';
 
 const version = '15.1.0';
 
 export default (options: Options): Rule => async (host: Tree, context: SchematicContext) => {
     context.logger.info(`Applying migration for Ignite UI for Angular to version ${version}`);
 
+    const { HtmlParser } = await nativeImport('@angular/compiler') as typeof import('@angular/compiler');
     const update = new UpdateChanges(__dirname, host, context);
+    const cardsToMigrate = new Set<any>();
+    const CARD_ACTIONS = ['igx-card-actions'];
+    const prop = ['igxButton'];
+    const changes = new Map<string, FileChange[]>();
+    
+    const applyChanges = () => {
+      for (const [path, change] of changes.entries()) {
+        let buffer = host.read(path).toString();
+
+        change.sort((c, c1) => c.position - c1.position)
+          .reverse()
+          .forEach(c => buffer = c.apply(buffer));
+
+        host.overwrite(path, buffer);
+      }
+    };
+
+    const addChange = (path: string, change: FileChange) => {
+        if (changes.has(path)) {
+          changes.get(path).push(change);
+        } else {
+          changes.set(path, [change]);
+        }
+    };
+
+    const getChildren = (parent: Element, buttons: any[], icons: any[]) => {
+        const cardButtons = parent.children.filter(column => (column as Element).name === 'button');
```

Comment:
I see a problem here. Elements other than just `button` can also have the `igxButton` directive applied to them. For instance `span` or `a` elements can have it. So filtering by `name` is not exhaustive. We should be looking for all children that have the `igxButton` attribute, but not the ones that have `igxButton="icon"`.

Elements with the `igxButton="icon"` should go in the cardIcons collection.